toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/OfflineRepository.kt

```diff
@@ -548,8 +548,7 @@ public object OfflineRepository {
                     when (workInfo.state) {
                         // if work completed successfully
                         WorkInfo.State.SUCCEEDED -> {
-                            // TODO: Wire in status handler
-                            // onDemandMapAreasState.updateStatus(Status.Downloaded)
+                            onDemandMapAreasState.updateStatus(com.arcgismaps.toolkit.offline.ondemand.Status.Downloaded)
```

Comment:
Should we refactor the two `Status` classes to `OnDemandStatus` and `PreplannedStatus` for clarity here?  Or optionally, if we should make `Status` an interface used by on-demand & preplanned?
 
```suggestion
                            onDemandMapAreasState.updateStatus(OnDemandStatus.Downloaded)
```

Reply:
We need to clean up the Status for OnDemand, It uses totally different values compared to Preplanned(it does not take into account any of the packaging statuses. In that regard I don't know if the interface makes sense. I can rename them for now. We can refactor after we have tested some scenarios for onDemand